Solve Sequential Promise Problem Day 14

unikuam opened the pull request:
I pushed this solution earlier in "exercises" branch by mistake so I am pushing it again in a different branch to create a separate PR.

Here, a chain of promises is created using for loop and then resolved the same chain.

unikuam committed:
Solve Sequential Promise Problem Day 14

I pushed this solution earlier in exercises branch by mistake so I am pushing it again in different branch.

ArfatSalman commented on js-exercises/sequential-promise/sequentialPromise.js:
```diff
@@ -0,0 +1,13 @@
+const sequentialPromise = promises => {
```
Are you accepting promises or promise-generating functions? You are executing `promises` which means they are functions, but the name suggests they are an array of promises.

unikuam replied:
I am accepting an array of promises so the name is incorrect. It should be promisesArray. I'll correct it.

ArfatSalman commented on js-exercises/sequential-promise/sequentialPromise.js:
```diff
@@ -0,0 +1,13 @@
+const sequentialPromise = promises => {
+  let promise = promises[0]();
+  for (let i = 1; i < promises.length; i++) {
+    if (typeof promises[i] !== 'function') {
```
This would throw an error before any of the promises can actually execute.

ArfatSalman replied:
Do this check before you are executing the first promise.

ArfatSalman commented on js-exercises/sequential-promise/sequentialPromise.js:
```diff
@@ -0,0 +1,13 @@
+    }
+    promise = promise.then(promises[i]);
+  }
+  return Promise.resolve(promise);
```
You need to resolve with the collected values and not the final promise.

ArfatSalman commented on js-exercises/sequential-promise/sequentialPromise.js:
```diff
@@ -0,0 +1,13 @@
+    if (typeof promises[i] !== 'function') {
+      throw new Error(`Expected array of functions as arguments, got ${typeof promises[i]}`)
+    }
+    promise = promise.then(promises[i]);
```
Try doing without using `let`

unikuam replied:
Okay, I'll without let.

ArfatSalman commented on js-exercises/sequential-promise/sequentialPromise.test.js:
```diff
@@ -0,0 +1,45 @@
+import { sequentialPromise } from './sequentialPromise';
+
+describe('sequentialPromise', () => {
+  test('should resolve 3 promises in sequence and return a promise', async () => {
```
Add more test cases.

unikuam replied:
okay, I'll add the same.

unikuam committed:
Add test cases & correct the implementation (in progress)

I made changes in the code according to arfat's reviews. Added more test cases. In this code, the results at line number 32 is giving correct result but the "resolve(result)" is not being returned with these values so I am working on the same.

unikuam committed:
Change global var declaration to const from let
